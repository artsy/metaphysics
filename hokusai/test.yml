version: "2"
services:
  metaphysics:
    command: env NODE_OPTIONS=--max_old_space_size=4096 yarn ci
    extends:
      file: build.yml
      service: metaphysics
    environment:
      - MEMCACHED_URL=metaphysics-memcached:11211
      - CI_PULL_REQUEST=$CI_PULL_REQUEST
      - CIRCLE_PULL_REQUEST=$CIRCLE_PULL_REQUEST
      - CIRCLE_BUILD_NUM=$CIRCLE_BUILD_NUM
      - ENABLE_CONSIGNMENTS_STITCHING=true
    env_file: ../.env.test
    depends_on:
      - metaphysics-memcached
  metaphysics-memcached:
    image: memcached
    ports:
      - 11211:11211
