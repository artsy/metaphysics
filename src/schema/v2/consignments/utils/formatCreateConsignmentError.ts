import {
  GraphQLInt,
  GraphQLNonNull,
  GraphQLObjectType,
  GraphQLString,
} from "graphql"
import { formatGravityError } from "lib/gravityErrorHandler"
import { ResolverContext } from "types/graphql"

export const ConsignmentInquiryMutationErrorName =
  "ConsignmentInquiryMutationError"

export const ConsignmentInquiryErrorType = new GraphQLObjectType<
  any,
  ResolverContext
>({
  name: ConsignmentInquiryMutationErrorName,
  fields: () => ({
    type: {
      type: GraphQLString,
    },
    message: {
      type: new GraphQLNonNull(GraphQLString),
    },
    error: {
      type: GraphQLString,
    },
    statusCode: {
      type: GraphQLInt,
    },
  }),
})

export const formatCreateConsignmentError = (error) => {
  return formatGravityError(error)
}
