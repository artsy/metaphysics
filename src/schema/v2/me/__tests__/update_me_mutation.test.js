/* eslint-disable promise/always-return */
import { runAuthenticatedQuery } from "schema/v2/test/utils"
import gql from "lib/gql"

describe("UpdateMeMutation", () => {
  it("updates the user profile and returns its new data payload", async () => {
    const mutation = gql`
      mutation {
        updateMyUserProfile(
          input: {
            clientMutationId: "1232"
            collectorLevel: 1
            bio: "A very long story"
            iconUrl: "https://gggg.s3.amazonaws.com/thekey"
            location: { address: "123 my street" }
            phone: "1234890"
            priceRangeMax: 1000000000000
            priceRangeMin: -1
            privacy: 1
            profession: "Juggler"
            receiveLotOpeningSoonNotification: false
            receiveNewSalesNotification: false
            receiveNewWorksNotification: true
            receiveOutbidNotification: false
            receivePromotionNotification: false
            receivePurchaseNotification: false
            receiveSaleOpeningClosingNotification: false
            shareFollows: false
          }
        ) {
          user {
            name
            phone
            location {
              city
              address
            }
            priceRange
            receivePurchaseNotification
            receiveOutbidNotification
            receiveLotOpeningSoonNotification
            receiveSaleOpeningClosingNotification
            receiveNewWorksNotification
            receiveNewSalesNotification
            receivePromotionNotification
          }
          userOrError {
            ... on UpdateMyProfileMutationSuccess {
              user {
                internalID
              }
            }
            ... on UpdateMyProfileMutationFailure {
              mutationError {
                type
                fieldErrors {
                  name
                  message
                }
              }
            }
          }
          me {
            name
          }
        }
      }
    `

    const mockUpdateMeLoader = jest.fn().mockReturnValue(
      Promise.resolve({
        id: "106",
        location: { address: "123 my street" },
        name: "andy-warhol",
        phone: "1234890",
        price_range: "-1:1000000000000",
        receive_lot_opening_soon_notification: false,
        receive_new_sales_notification: false,
        receive_new_works_notification: true,
        receive_outbid_notification: false,
        receive_promotion_notification: false,
        receive_purchase_notification: false,
        receive_sale_opening_closing_notification: false,
      })
    )

    const mockUpdateCollectorProfileIconLoader = jest.fn()

    const context = {
      meLoader: () =>
        Promise.resolve({
          id: "106",
          location: { address: "123 my street" },
          name: "andy-warhol",
          phone: "1234890",
          price_range: "-1:1000000000000",
          receive_lot_opening_soon_notification: true,
          receive_new_sales_notification: true,
          receive_new_works_notification: true,
          receive_outbid_notification: true,
          receive_promotion_notification: true,
          receive_purchase_notification: true,
          receive_sale_opening_closing_notification: true,
        }),
      updateMeLoader: mockUpdateMeLoader,
      updateCollectorProfileIconLoader: mockUpdateCollectorProfileIconLoader,
    }

    await runAuthenticatedQuery(mutation, context).then((data) => {
      expect(data).toMatchSnapshot()
    })

    expect(mockUpdateMeLoader).toBeCalledWith({
      client_mutation_id: "1232",
      collector_level: 1,
      bio: "A very long story",
      location: { address: "123 my street" },
      phone: "1234890",
      price_range_max: 1000000000000,
      price_range_min: -1,
      privacy: 1,
      profession: "Juggler",
      receive_lot_opening_soon_notification: false,
      receive_new_sales_notification: false,
      receive_new_works_notification: true,
      receive_outbid_notification: false,
      receive_promotion_notification: false,
      receive_purchase_notification: false,
      receive_sale_opening_closing_notification: false,
      share_follows: false,
    })

    expect(mockUpdateCollectorProfileIconLoader).toBeCalledWith({
      source_bucket: "gggg",
      source_key: "thekey",
    })

    expect.assertions(3)
  })

  it("does not update the icon when source url is not from s3", async () => {
    const mutation = gql`
      mutation {
        updateMyUserProfile(
          input: {
            clientMutationId: "1232"
            collectorLevel: 1
            bio: "A very long story"
            iconUrl: "https://gggg.notS3.com/thekey"
            privacy: 1
          }
        ) {
          user {
            name
            phone
          }
          userOrError {
            ... on UpdateMyProfileMutationSuccess {
              user {
                internalID
              }
            }
            ... on UpdateMyProfileMutationFailure {
              mutationError {
                type
                fieldErrors {
                  name
                  message
                }
              }
            }
          }
          me {
            name
          }
        }
      }
    `

    const mockUpdateMeLoader = jest.fn().mockReturnValue(
      Promise.resolve({
        id: "106",
        location: { address: "123 my street" },
        name: "andy-warhol",
        phone: "1234890",
        price_range: "-1:1000000000000",
        receive_lot_opening_soon_notification: false,
        receive_new_sales_notification: false,
        receive_new_works_notification: true,
        receive_outbid_notification: false,
        receive_promotion_notification: false,
        receive_purchase_notification: false,
        receive_sale_opening_closing_notification: false,
      })
    )

    const mockUpdateCollectorProfileIconLoader = jest.fn()

    const context = {
      meLoader: () =>
        Promise.resolve({
          id: "106",
          name: "andy-warhol",
        }),
      updateMeLoader: mockUpdateMeLoader,
      updateCollectorProfileIconLoader: mockUpdateCollectorProfileIconLoader,
    }

    await runAuthenticatedQuery(mutation, context).then((data) => {
      expect(data).toMatchSnapshot()
    })

    expect(mockUpdateMeLoader).toBeCalledWith({
      client_mutation_id: "1232",
      collector_level: 1,
      bio: "A very long story",
      privacy: 1,
    })

    expect(mockUpdateCollectorProfileIconLoader).not.toHaveBeenCalled()
  })
})
