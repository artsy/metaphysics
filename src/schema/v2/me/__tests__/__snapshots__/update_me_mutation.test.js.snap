// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateMeMutation does not update the icon when source url is not from s3 1`] = `
Object {
  "updateMyUserProfile": Object {
    "me": Object {
      "name": "andy-warhol",
    },
    "user": Object {
      "name": "andy-warhol",
      "phone": "1234890",
    },
    "userOrError": Object {
      "user": Object {
        "internalID": "106",
      },
    },
  },
}
`;

exports[`UpdateMeMutation updates the user profile and returns its new data payload 1`] = `
Object {
  "updateMyUserProfile": Object {
    "me": Object {
      "name": "andy-warhol",
    },
    "user": Object {
      "location": Object {
        "address": "123 my street",
        "city": null,
      },
      "name": "andy-warhol",
      "phone": "1234890",
      "priceRange": "-1:1000000000000",
      "receiveLotOpeningSoonNotification": false,
      "receiveNewSalesNotification": false,
      "receiveNewWorksNotification": true,
      "receiveOutbidNotification": false,
      "receivePromotionNotification": false,
      "receivePurchaseNotification": false,
      "receiveSaleOpeningClosingNotification": false,
    },
    "userOrError": Object {
      "user": Object {
        "internalID": "106",
      },
    },
  },
}
`;
